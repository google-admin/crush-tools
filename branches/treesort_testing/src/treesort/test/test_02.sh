test_number=02
description="three column sort"

subtest_desc=("CC,BB,DD" 
							"AA,BB,CC"
							"DD,CC,BB"
							"4,2,1 with indexes"
							'^AA,BB,CC'
							'^AA,^BB,CC'
							'^AA,^BB,^CC'
							'1,^2,^3 with indexes'
							'AA,BB,^CC'
							'1,^2,3 with indexes'
							)
subtest_opts=(
'-K CC,BB,DD'
'-K AA,BB,CC'
'-K DD,CC,BB'
'-k 4,2,1 -p'
'-K ^AA,BB,CC'
'-K ^AA,^BB,CC'
'-K ^AA,^BB,^CC'
'-k 1,^2,^3 -p'
'-K AA,BB,^CC'
'-k 1,^2,3 -p'
)

for subtest in `seq 0 9`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test2.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
