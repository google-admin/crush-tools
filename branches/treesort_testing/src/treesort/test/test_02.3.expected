DDþBBþAAþCC
11.9550þ5-dmH-yGymþ120þ2009-10-29
11.9550þOxNCXf_QO5þ222þ2009-10-15
11.9550þPb0jUNXPGFþ704þ2009-10-21
11.9550þV-dWH8XhzNþ704þ2009-10-15
11.9550þZeP1N6ZHo8þ222þ2009-10-09
11.9550þlAkjpi7B0uþ704þ2009-10-21
11.9550þoj-npio3Cqþ119þ2009-10-18
11.9550þqdigsUU1EHþ222þ2009-10-21
13.3454þBuPJhoNfZYþ692þ2009-10-21
13.3454þD6DavOvkp5þ119þ2009-10-21
13.3454þDx5ojQn-G6þ379þ2009-10-19
13.3454þHF90x6_QiBþ222þ2009-10-23
13.3454þOypnZVbJOYþ876þ2009-10-15
13.3454þ_Dh-EwmwHUþ222þ2009-10-29
13.3454þbNfKeDJp35þ120þ2009-10-15
13.3454þdCzebYQfiHþ222þ2009-10-18
13.3454þeGWkG9dnHHþ790þ2009-10-29
13.3454þgbwjbbmOR5þ704þ2009-10-25
13.3454þmdg21XYvE4þ379þ2009-10-18
14.6088þ1aRiJ5zvaUþ379þ2009-10-15
14.6088þQ99ceOGe58þ119þ2009-10-15
14.6088þQEIlvFBOAEþ428þ2009-10-21
14.6088þQHJtym0emOþ790þ2009-10-19
14.6088þY0_LegXodZþ120þ2009-10-18
14.6088þooYbIsNctcþ692þ2009-10-21
14.6088þqv6ulolvVmþ876þ2009-10-18
16.2961þ3Abgi5FvpKþ692þ2009-10-18
16.2961þAnKSv96_ykþ525þ2009-10-29
16.2961þFJFwWbUCEgþ525þ2009-10-18
16.2961þGkijsWrtDmþ222þ2009-10-25
16.2961þJw-SmuJJwHþ428þ2009-10-29
16.2961þRHEIKKC-eNþ876þ2009-10-06
16.2961þU_DsBKeZBOþ790þ2009-10-15
16.2961þXdQde3c2LRþ120þ2009-10-29
16.2961þab_ALFxdBlþ704þ2009-10-23
16.2961þkX69CHRCpFþ119þ2009-10-29
16.2961þvT9F16VWxuþ876þ2009-10-02
16.2961þw-DsqhmkP4þ428þ2009-10-06
19.8086þ0J2H9eMxz5þ428þ2009-10-19
19.8086þ1Qbw9NbEK2þ120þ2009-10-19
19.8086þCwEfrzVzlNþ704þ2009-10-29
19.8086þMP3BenV_PWþ525þ2009-10-02
19.8086þMk8GiTvT-2þ222þ2009-10-21
19.8086þN1CuDuqLsqþ120þ2009-10-25
19.8086þV_Srbg1Xgkþ525þ2009-10-09
19.8086þWxgda_Rz1uþ120þ2009-10-23
19.8086þcOA-GflWf9þ120þ2009-10-02
19.8086þqHbD7IgvBsþ119þ2009-10-15
20.0623þH3laCVXVqFþ525þ2009-10-29
20.0623þQOhQYF2L4Uþ120þ2009-10-18
20.0623þQddzqzZhvaþ222þ2009-10-19
20.0623þR0lHfvqeE2þ222þ2009-10-18
20.0623þSEiwhaVJcvþ790þ2009-10-25
20.0623þUH3YCtEsnEþ379þ2009-10-21
20.0623þgplXXJQCbDþ692þ2009-10-23
20.0623þi4Vn-mjUX_þ379þ2009-10-25
20.0623þs5FX4ERZP8þ119þ2009-10-19
20.0623þtH_urSAL6Iþ379þ2009-10-15
21.4784þ-_WgFpelwTþ119þ2009-10-15
21.4784þ-l43yW3QQXþ379þ2009-10-19
21.4784þ6RIZ_xP77Oþ876þ2009-10-23
21.4784þDMqu6KjLMnþ428þ2009-10-09
21.4784þK26asHqM1Bþ692þ2009-10-25
21.4784þMLySuT_6L-þ119þ2009-10-25
21.4784þQOa1zDDXrcþ692þ2009-10-19
21.4784þZ_RNo9lPThþ222þ2009-10-09
21.4784þZs9uH9wqwvþ120þ2009-10-19
21.4784þ_6GXpFrx7Mþ704þ2009-10-09
21.4784þ_okpFO60P5þ120þ2009-10-19
21.4784þghqsjQH7CAþ790þ2009-10-25
21.4784þkC3oWoj1_8þ692þ2009-10-25
21.4784þnLKFOKtBt5þ876þ2009-10-19
21.4784þpfetgY4yYtþ525þ2009-10-29
21.4784þyj8FAR6sT8þ790þ2009-10-19
23.1071þNrNOPtKMh2þ222þ2009-10-09
23.1071þS-BUVAIarsþ120þ2009-10-02
23.1071þVjx-O1uUiyþ119þ2009-10-06
23.1071þZ971TGdirGþ876þ2009-10-21
23.1071þ_mNRE6q64Dþ525þ2009-10-29
23.1071þbWuILAzu5Lþ379þ2009-10-06
23.1071þcEXl50cdsGþ790þ2009-10-02
23.1071þi0-lOQWHXeþ428þ2009-10-18
23.1071þj-OdwoZTzmþ692þ2009-10-21
23.1071þpPrA9-ibvPþ790þ2009-10-19
23.1071þuGBjnSxQwIþ692þ2009-10-19
25.4415þ0CjQNzoprAþ790þ2009-10-21
25.4415þ8ZDolJWFhKþ120þ2009-10-19
25.4415þNqFghFb16Pþ119þ2009-10-19
25.4415þOcgRMuIUVWþ120þ2009-10-09
25.4415þQg_963dGw2þ428þ2009-10-23
25.4415þccfVWBl1Ifþ379þ2009-10-15
25.4415þczHP0Y09gOþ119þ2009-10-18
25.4415þn1pETdd_e2þ790þ2009-10-25
43.1227þ-ca4VazCCoþ525þ2009-10-19
43.1227þ-ynhLNQjXnþ525þ2009-10-18
43.1227þ8HPzHGZLDfþ222þ2009-10-02
43.1227þPKndexdB5Yþ119þ2009-10-06
43.1227þl0sIDKi1Mnþ222þ2009-10-09
43.1227þpUbcEzSLg5þ379þ2009-10-15
43.1227þzHaJcq1brbþ379þ2009-10-21
